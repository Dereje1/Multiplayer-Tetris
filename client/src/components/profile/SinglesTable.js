import * as React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
/* MUI */
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';

export default function SinglesTable({rows}) {
  return (
    <TableContainer component={Paper}>
      <Table sx={{ minWidth: 650 }} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Date</TableCell>
            <TableCell align="center">Lines Cleared</TableCell>
            <TableCell align="center">Level Reached</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.map((row) => (
            <TableRow
              key={row._id}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <TableCell component="th" scope="row">
                {moment.utc(row.createdAt).format('MMMM Do YYYY, h:mm:ss a')}
              </TableCell>
              <TableCell align="center">{row.linesCleared}</TableCell>
              <TableCell align="center">{row.levelReached}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

SinglesTable.propTypes = {
  rows:PropTypes.arrayOf(PropTypes.object).isRequired
}