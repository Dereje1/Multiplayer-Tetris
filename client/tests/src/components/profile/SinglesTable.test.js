import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SinglesTable from '../../../../src/components/profile/SinglesTable'
import { stubProfile } from '../../../stub'

describe('The Single games table', () => {
    test('will render', () => {
        const wrapper = shallow(<SinglesTable rows={stubProfile.userProfileResponse.singleStats} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})