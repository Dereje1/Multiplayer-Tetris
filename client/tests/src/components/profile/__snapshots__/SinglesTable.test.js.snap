// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Single games table will render 1`] = `
<ForwardRef(TableContainer)
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "propTypes": Object {
        "children": [Function],
        "className": [Function],
        "classes": [Function],
        "component": [Function],
        "elevation": [Function],
        "square": [Function],
        "sx": [Function],
        "variant": [Function],
      },
      "render": [Function],
    }
  }
>
  <ForwardRef(Table)
    aria-label="simple table"
    sx={
      Object {
        "minWidth": 650,
      }
    }
  >
    <ForwardRef(TableHead)>
      <ForwardRef(TableRow)>
        <ForwardRef(TableCell)>
          Date
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
        >
          Lines Cleared
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
        >
          Level Reached
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
    </ForwardRef(TableHead)>
    <ForwardRef(TableBody)>
      <ForwardRef(TableRow)
        key="62e3f935616501f9bd180a71"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          July 29th 2022, 3:13:57 pm
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
        >
          49
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
        >
          9
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="62dfd050d218114e5c901c43"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          July 26th 2022, 11:30:24 am
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
        >
          60
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)
          align="center"
        >
          12
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
    </ForwardRef(TableBody)>
  </ForwardRef(Table)>
</ForwardRef(TableContainer)>
`;
